lisa-utils/src/main/scala/lisa/utils/Parser.scala

```diff
@@ -10,37 +10,39 @@ import scallion.util.Unfolds.unfoldRight
 import silex.*
 
 object Parser {
-  enum Notation {
-    case Prefix, Infix
-  }
-
-  abstract class Label(val id: String, val notation: Notation)
-
-  case class InfixLabel(override val id: String) extends Label(id, Notation.Infix)
+  //////////////////////////////// EQUIVALENT LABELS, INFIX CONTROL ///////////////////////////////
+  case class CanonicalId(print: String, internal: String)
 
-  case class PrefixLabel(override val id: String) extends Label(id, Notation.Prefix)
+  def equivalentLabelsToMap(equivalentLabels: List[String], print: String, internal: String): Map[String, CanonicalId] =
+    (print :: internal :: equivalentLabels).map(_ -> CanonicalId(print, internal)).toMap
 
-  case class CanonicalLabel(print: Label, internal: Label)
+  // TODO: grow the mapping as we discover / introduce more synonyms
+  // The mapping stores the information about synonymous strings for the same id, mapping a string to its canonical
+  // form (the id that is used to construct the `ConstantLabel`) and the printable form (the preferred synonym to
+  // display in strings).
+  private val SynonymToCanonical: Map[String, CanonicalId] =
+    equivalentLabelsToMap("elem" :: "in" :: Nil, "∊", "elem") ++
+      equivalentLabelsToMap("subset_of" :: "subset" :: Nil, "⊆", "subset_of") ++
+      equivalentLabelsToMap("sim" :: "same_cardinality" :: Nil, "≈", "same_cardinality")
```

Comment:
I believe $\approx$ is \approx in latex, whereas $\sim$ is \sim. So, maybe internally we should not use `sim`?

Reply:
Makes sense; I added "sim" because the corresponding `ConstantPredicateLabel` is introduced as `final val sim = ConstantPredicateLabel("same_cardinality", 2)` and as pointed out by @sankalpgambhir, it would indeed be confusing to users that they can write `sim(x, y)` in Scala code but not in the string representation of the theorem.

Reply:
We can leave both for now. At some point we should discuss if we want to change the strings and Scala identifiers for some of these.

Reply:
So should I leave `sim` as a synonym or remove it?

Reply:
Can we please focus on something that is adaptable and generic?
Maybe at some point in the future we will have to actually hard code in the parser some set-theory related things like representation or functions, or lists, but for regular symbol, let's just factor them out of the parser.

Reply:
@SimonGuilloud moving things around may indeed be the right thing to do, so you want to move a constant string from one place in code base to another, sure. I should note that right now, if I want to understand what our parser does, I look into the parser file. Once it is all generic and parameterized, then I need to see what the places are where parser is called and how these arguments are created. It's the usual trade off. Try understanding Stainless syntax trees to see what happens when everything is parameterized and path dependent.

Reply:
@SimonGuilloud do you have a specific solution in mind?

Reply:
I did not say moving things arround, I said provide an interface and the complements to that interface, possibly non-static/library dependent, separately. But now focus on the mechanical part and outsource things  that need to be outrsourced, possibly with a placeolder hardcoded set/map for now.
But discussing which symbol is used for which label exactly is largeley irrelevant at this stage.